src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int hashCode = (key != null) ? key.hashCode() : 0;
+        int insertIndex = Math.abs(hashCode) % table.length;
+
+        Node<K, V> newNode = new Node<>(key, value, null);
+
+        Node<K, V> current = table[insertIndex];
+        while (current != null) {
+            if ((current.key == null && key == null)
+                    || (current.key != null && current.key.equals(key))) {
```

Comment:
```suggestion
            if ((current.key == key)
                    || (current.key != null && current.key.equals(key))) {
```
extract repeated logic to a private method